store/badger.go

```diff
@@ -189,3 +213,36 @@ func (i *BadgerIterator) Discard() {
 	i.iter.Close()
 	i.txn.Discard()
 }
+
+// memoryEfficientBadgerConfig sets badger configuration parameters to reduce memory usage, specially during compactions to avoid memory spikes that causes OOM.
+// based on https://github.com/celestiaorg/celestia-node/issues/2905
+func memoryEfficientBadgerConfig(path string, syncWrites bool) *badger.Options {
+	opts := badger.DefaultOptions(path) // this must be copied
+	// SyncWrites is a configuration option in Badger that determines whether writes are immediately synced to disk or no.
+	// If set to true it writes to the write-ahead log (value log) are synced to disk before being applied to the LSM tree.
+	opts.SyncWrites = syncWrites
+	// default 64mib => 0 - disable block cache
+	// BlockCacheSize specifies how much data cache should hold in memory.
+	// It improves lookup performance but increases memory consumption.
+	// Not really necessary if disabling compression
+	opts.BlockCacheSize = 0
+	// compressions reduces storage usage but increases memory consumption, specially during compaction
+	opts.Compression = options.None
```

Comment:
no compression at all?
maybe we need to separate between sequencer/full node/archive

Reply:
the problem with dymint store is not the storage space but the memory consumption when compacting. its true its not optimal (the size it can be doubled when using no compression) but taking into account that in an unpruned node the application db space can be more than 10x the dymint storage space, it does not seem to be the issue. also this is a short-term solution to avoid oom issues in nodes. long-term solution will probably be replace badger by a more optimal store for dymint requirements.